src/common/dsp/SurgeVoice.cpp

```diff
@@ -694,29 +694,32 @@ bool SurgeVoice::process_block(QuadFilterChainState& Q, int Qe)
    float ktrkroot = 60;
    float drift = localcopy[scene->drift.param_id_in_scene].f;
 
+   auto pb = modsources[ms_pitchbend]->output;
+   if (pb > 0)
+      pb *= (float)scene->pbrange_up.val.i;
+   else
+      pb *= (float)scene->pbrange_dn.val.i;
+
+   // recalculate pitch state but without state.pkey, so that the residue can be used to add scene pitch/octave/pitch bend
+   // and associated modulations to non-keytracked oscillators
+   auto tempPitch = pb + localcopy[pitch_id].f * (scene->pitch.extend_range ? 12.f : 1.f) + (octaveSize * localcopy[octave_id].i);
```

Comment:
Since there's a couple of brace elisions which we should fix I think it also makes sense to choose a better name than 'tempPitch'. I was thinking "sceneAndPBPitchAdj" or something. So the code below reads as 'ktkroot + sceneAndPBPitchAdj". Or some such.

Reply:
Makes sense, I also moved it to member variable of voice state. Code is clearer now!